rust/src/connection/server_connection.rs

```diff
@@ -134,37 +132,43 @@ impl ServerConnection {
     }
 
     #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
-    pub(crate) async fn database_exists(&self, database_name: String) -> crate::Result<bool> {
-        match self.request(Request::DatabasesContains { database_name }).await? {
-            Response::DatabasesContains { contains } => Ok(contains),
+    pub(crate) async fn all_databases(&self) -> crate::Result<Vec<DatabaseInfo>> {
+        match self.request(Request::DatabasesAll).await? {
+            Response::DatabasesAll { databases } => Ok(databases),
             other => Err(InternalError::UnexpectedResponseType { response_type: format!("{other:?}") }.into()),
         }
     }
 
     #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
-    pub(crate) async fn create_database(&self, database_name: String) -> crate::Result<DatabaseInfo> {
-        match self.request(Request::DatabaseCreate { database_name }).await? {
-            Response::DatabaseCreate { database } => Ok(database),
+    pub(crate) async fn get_database_replicas(&self, database_name: String) -> crate::Result<DatabaseInfo> {
+        match self.request(Request::DatabaseGet { database_name }).await? {
+            Response::DatabaseGet { database } => Ok(database),
             other => Err(InternalError::UnexpectedResponseType { response_type: format!("{other:?}") }.into()),
         }
     }
 
     #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
-    pub(crate) async fn get_database_replicas(&self, database_name: String) -> crate::Result<DatabaseInfo> {
-        match self.request(Request::DatabaseGet { database_name }).await? {
-            Response::DatabaseGet { database } => Ok(database),
+    pub(crate) async fn database_exists(&self, database_name: String) -> crate::Result<bool> {
```

Comment:
I'd rename it to `contains_database` similary to `contains_user`

Reply:
Or the other way around

Reply:
Fixed in 32a91059c387c866c4c00443691d649347fbb4dd.